### ISSUE TITLE
Duplicated subtitles every time refresh is forced

### ISSUE THREAD

**Issue by BaNkIck** [pannal/Sub-Zero#29]
Every time I force an update on Plex, even on a single episode, it downloads the same subtitles again.

I have configured to download subtitles for three different languages (English, Catalan and Spanish). Currently, I have noticed this behavior on a particular episode that it only finds them in one language (English). Maybe (just maybe, haven't tested it yet) it keeps downloading the same subtitle only when subtitles are missing for one of the required languages.

I have the RC-3 version of the bundle.

Here is the xml of the episode (as Plex shows it):

```xml
<?xml version="1.0" encoding="UTF-8"?>
<MediaContainer size="1" allowSync="1" identifier="com.plexapp.plugins.library" librarySectionID="1" librarySectionTitle="TV Shows" librarySectionUUID="1c284e3f-bc96-458d-8c6a-71558aac53e7" mediaTagPrefix="/system/bundle/media/flags/" mediaTagVersion="1443813486">
    <Video ratingKey="3467" key="/library/metadata/3467" parentRatingKey="3281" grandparentRatingKey="574" guid="com.plexapp.agents.thetvdb://95011/7/3?lang=en" type="episode" title="The Closet Case" titleSort="Closet Case" grandparentKey="/library/metadata/574" parentKey="/library/metadata/3281" grandparentTitle="Modern Family" contentRating="TV-PG" summary="{actual summary}" index="3" parentIndex="7" rating="8.0" year="2015" thumb="/library/metadata/3467/thumb/1444418489" art="/library/metadata/574/art/1444314049" parentThumb="/library/metadata/3281/thumb/1444314049" grandparentThumb="/library/metadata/574/thumb/1444314049" grandparentArt="/library/metadata/574/art/1444314049" grandparentTheme="/library/metadata/574/theme/1444314049" duration="1262387" originallyAvailableAt="2015-10-07" addedAt="1444313492" updatedAt="1444418489" chapterSource="">
        <Media videoResolution="720" id="6490" duration="1262387" bitrate="3572" width="1280" height="718" aspectRatio="1.78" audioChannels="6" audioCodec="ac3" videoCodec="h264" container="mkv" videoFrameRate="24p">
            <Part accessible="1" exists="1" id="6510" key="/library/parts/6510/file.mkv" duration="1262387" file="{path to actual file}" size="563649438" container="mkv">
                <Stream id="18322" streamType="1" codec="h264" index="0" bitrate="3117" language="English" languageCode="eng" bitDepth="8" cabac="1" chromaSubsampling="4:2:0" codecID="V_MPEG4/ISO/AVC" colorSpace="yuv" duration="1262387" frameRate="23.976" frameRateMode="cfr" hasScalingMatrix="0" height="718" level="41" profile="high" refFrames="5" scanType="progressive" width="1280" />
                <Stream id="18323" streamType="2" selected="1" codec="ac3" index="1" channels="6" bitrate="384" audioChannelLayout="5.1(side)" bitDepth="16" bitrateMode="cbr" codecID="A_AC3" dialogNorm="-24" duration="1262368" samplingRate="48000" />
                <Stream id="19041" key="/library/streams/19041" streamType="3" codec="srt" language="English" languageCode="eng" format="" />
                <Stream id="19039" key="/library/streams/19039" streamType="3" selected="1" codec="srt" language="English" languageCode="eng" format="" />
                <Stream id="19040" key="/library/streams/19040" streamType="3" codec="srt" language="English" languageCode="eng" format="" />
            </Part>
        </Media>
        <Writer id="1450" tag="Paul Corrigan" />
        <Writer id="1451" tag="Brad Walsh" />
        <Director id="1473" tag="Beth McCarthy-Miller" />
        <Extras size="0"></Extras>
    </Video>
</MediaContainer>
```

The Subtitles.xml, inside the Media folder of this particular episode:

```xml
<?xml version='1.0' encoding='utf8'?>
<Subtitles>
  <Language code="ca"/>
  <Language code="en">
    <Subtitle media="com.plexapp.agents.subliminal_2225ef11fa10e0bd0edca3f4eabfcc1c536fa35c.srt" codec="srt" name="http://www.opensubtitles.org/en/subtitles/6331679/sid-c7m860o0qg9fuun80ugkg9p1f4/modern-family-the-closet-case-en"/>
    <Subtitle media="com.plexapp.agents.subliminal_f30473dc1ace55418cbf9b3895817d3d07c43942.srt" codec="srt" name="http://www.opensubtitles.org/en/subtitles/6331679/sid-djbb1blugdprf6c1aolbldbp43/modern-family-the-closet-case-en"/>
    <Subtitle media="com.plexapp.agents.subliminal_57bac6b947a81a67ce91cce9b6862dd464f198f2.srt" codec="srt" name="http://www.opensubtitles.org/en/subtitles/6331679/sid-030hf05o6k5sojjj2mvurroiu3/modern-family-the-closet-case-en"/>
  </Language>
</Subtitles>
```

And a screenshot:

![captura de pantalla 2015-10-09 21 21 56](https://cloud.githubusercontent.com/assets/843170/10403468/195cddda-6ecc-11e5-91de-917b294523cf.png)

**Comment by pannal** [comment]
Oh that sounds weird.

To sum it up again, you've got a Video and you have 3 subtitle language configured in subliminal, right?

And it doesn't download any of the other two languages if one language's subtitle is already there?
Are you using metadata as storage or an external folder?

**Comment by BaNkIck** [comment]
Sorry I didn't make myself clear the first time.

I have 3 languages configured. Subliminal cannot find subtitles for this particular episode in two of these languages yet (Catalan and Spanish). I know it because I have subliminal installed as standalone binary too and I have tested it directy from the command line (subliminal download -l en -l ca -l es {file.mkv}).

I'm using metadata storage.
Yes, it downloads the same subtitle over and over.
I haven't tried to switch to external storage (I'm gonna)
Sorry, I cannot find the log file. Where is it located?
And no, those two options aren't checked.

I'll play a bit with all these options you named and get back to you with the results.

Oh! "Cap" means "None" in Catalan :)

**Comment by pannal** [comment]
Info about the log file location: https://github.com/pannal/Subliminal.bundle/blob/master/README.md

That behaviour is intended. If you leave those two options unchecked, it will always download a found subtitle regardless of the fact that there already is one (embedded or external).

**Comment by pannal** [comment]
I guess the two options are a bit misleading.

Would renaming them to do the opposite help? Like
"Ignore embedded subtitles when searching" and "Ignore external subtitles (overwrite unconditionally on download)"

**Comment by chopeta** [comment]
I agree with the rename idea.

**Comment by BaNkIck** [comment]
I think the options are more clear the way they are now. The new proposed names confuse me more. Perhaps, what would be interesting is that these options are enabled by default (as they are right now, obviously, if you invert them, then they shouldn't be enabled by default).

Moving on to the actual issue. I have been playing a lot with different combinations of options. To sum it up, the problem only happens when using metadata storage. In this case, if I don't have any external subtitle or if a have some but not checked the "Scan: include external subtitles" option, it keeps re-downloading all available subtitles and adding them (not overwriting) to the episode. It seems the option "Scan: include embedded subtitles" doesn't work at all. Checked or unchecked, It always redownloads and adds subtitles.

I've discarded the problem was originated because subtitles for some of the configured languages weren't available. It doesn't really matter. I configured only the two available subtitles, forced a refresh on a particular episode and it re-download both available subtitles at that moment.

I also want to make clear the problem happens when I "Refresh" and not when I "Update" my library. Because of a translation mistake of PMS Media Manager, I thought they were the same option. When I switched PMS to English I noticed there are two different actions.

I'll attach the logs later today, sorry.
